### PR TITLE
config/server: add functions to check IDMap support in runtime

### PR DESCRIPTION
**Pull request by haircommander**
if we don't, we risk container creation failures or security holes from the runtime not correctly setting up idmaps

#### What type of PR is this?

/kind feature
#### Does this PR introduce a user-facing change?

```release-note
Check the runtime supports IDMap support before specifying 
```

### CODE REVIEW

**Comment by haircommander** [comment]
cc @rata

**Comment by sohankunkerkar** [comment]
Fixes https://github.com/cri-o/cri-o/issues/7262

File: server/container_create_linux.go [hunk 1]
```diff
@@ -1064,12 +1065,17 @@ func addOCIBindMounts(ctx context.Context, ctr ctrfactory.Container, mountLabel,
 			SelinuxRelabel: m.SelinuxRelabel,
 		})
 
+		uidMappings := getOCIMappings(m.UidMappings)
+		gidMappings := getOCIMappings(m.GidMappings)
+		if (uidMappings != nil || gidMappings != nil) && !idMapSupport {
+			return nil, nil, fmt.Errorf("IDMappings specified but runtime does not support them")
```
Comment (rata) [inline server/container_create_linux.go:1071]: This is shown like this in the kubectl describe output:
```
 Warning  Failed                  56s (x12 over 3m5s)  kubelet, 127.0.0.1  Error: IDMappings specified but runtime does not support them

```

I wonder if we can make it clearer for the user what they need to do? Maybe it is fine as it is, but maybe we can give more actionable info? I can't think of anything now, but:

```suggestion
			return nil, nil, fmt.Errorf("idmap mounts specified but OCI runtime does not support them, OCI runtime too old?")
```

File: server/container_create_linux_test.go [hunk 2]
```diff
@@ -159,6 +160,47 @@ func TestAddOCIBindsCGroupRW(t *testing.T) {
 	}
 }
 
+func TestAddOCIBindsErrorWithoutIDMap(t *testing.T) {
+	ctr, err := container.New()
+	if err != nil {
+		t.Error(err)
```
Comment (rata) [inline server/container_create_linux_test.go:166]: Do we want t.Fatal() here, so the execution doesn't continue?

```suggestion
		t.Fatal(err)
```

File: server/container_create_linux_test.go [hunk 3]
```diff
@@ -159,6 +160,47 @@ func TestAddOCIBindsCGroupRW(t *testing.T) {
...
+		},
+		Metadata: &types.ContainerMetadata{
+			Name: "testctr",
+		},
+	}, &types.PodSandboxConfig{
+		Metadata: &types.PodSandboxMetadata{
+			Name: "testpod",
+		},
+	}); err != nil {
+		t.Error(err)
```
Comment (rata) [inline server/container_create_linux_test.go:191]:
```suggestion
		t.Fatal(err)
```

**Review by rata — APPROVED** [review]
LGTM, thanks!

Left some minot comments, even if you want to act on them, I don't think I need to review again :)

**Comment by haircommander** [comment]
@cri-o/cri-o-maintainers PTAL 

This is related to https://github.com/kubernetes/website/pull/42778/ which is due tomorrow. Ideally we'd have a version of this PR backported to there and released as 1.28.1

**Comment by sohankunkerkar** [comment]
You may want to look into resolving some of the failing GitHub action tests.

**Review by sohankunkerkar — APPROVED** [review]
/lgtm
/retest

**Pull request merged**